bitnami/spark/templates/NOTES.txt

```diff
@@ -26,17 +34,37 @@
 
   First, obtain the master IP, to do that the service type must be NodePort or LoadBalancer. Run the following command to obtain the master IP and submit your application:
 
-{{ if or (eq "NodePort" .Values.service.type) (eq "LoadBalancer" .Values.service.type) }}
-  export MASTER_IP=$(kubectl get services | awk '/{{ include "spark.master.service.name" . }}/ { print $4 }')
-  docker run -ti bitnami/spark spark-submit --master spark://$MASTER_IP:{{ .Values.service.clusterPort }} \
+{{- if eq "NodePort" .Values.service.type }}
+
+  export SUBMIT_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[?(@.name=='cluster')].nodePort}" services {{ include "spark.master.service.name" . }})
+  export SUBMIT_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
+
+  docker run -ti bitnami/spark spark-submit --master spark://$SUBMIT_IP:$SUBMIT_PORT \
+      --class org.apache.spark.examples.SparkPi \
+      --deploy-mode cluster \
+      examples/jars/spark-examples_2.11-2.4.4.jar 1000
```

Comment:
Please don't add examples using docker. Adapt this example to use a "client" pod instead.

Reply:
Right, I did intentionally in order to prettify the output.

Reply:
done ;)